w3lib/http.py

```diff
@@ -27,7 +27,17 @@ def headers_raw_to_dict(headers_raw):
 
     if headers_raw is None:
         return None
-    headers = headers_raw.splitlines()
+
+    headers = []
+    for line in headers_raw.split(b'\r\n'):
```

Comment:
Why limit line boundaries to `\r\n`?

Reply:
The reason is that the specification (RFC 2616 cited above or RFC7230 which appears more recent) states that in HTTP messages, lines are folded specifically by CRLF (`\r\n`) and not by LF (`\n`). Not following this had lead to an issue where a header field value containing a LF (`\n`) was accidentally interpreted as a multiline field value. Without leading at least one space or tab char in the following line, however, the header ended up being interpreted as malformed and the lines were simply discarded when parsed out.

This is rather pathological and use of LF within a field value feels like a bad idea, but it doesn't appear to violate the specification so some HTTP messages with that structure exist.

Reply:
https://stackoverflow.com/a/6324184 suggests that `\n` should also be interpreted as a line terminator.

Reply:
I would leave `.splitlines()` nonetheless, though, given the current information we have. If later it turns out by allowing certain forms of line break we are misinterpreting header values, then we can change this accordingly.

Reply:
I propose that the function `headers_raw_to_dict` take in an optional parameter `strict` with the default value that does not change the existing behavior. This way, the code works as is, but would be ready for those who want a stricter interpretation of the HTTP standard via a simple switch.

I think exposing this switch to the users requires a similar change at the user-facing API, but that should be done in a separate PR, if desired.